stdlib/ballerina-builtin/src/main/ballerina/io/Package.md

```diff
@@ -0,0 +1,137 @@
+## Package Overview
+
+This package is designed to support input and output operations via channels in a canonical way, either in a blocking,
+or non-blocking manner.
+
+![architecture](resources/package-architecture.svg)
+
+### Channels
+A channel represents an I/O source or sink of some bytes, characters, or records that are opened for reading, or
+writing, respectively.
+### Byte channels
+The most primitive channel is the `ByteChannel` which reads and writes 8-bit bytes.
+
+```ballerina
+	// Open a file in read-write mode.
+io:ByteChannel byteChannel = io:openFile("some/file.txt", io:RW);
+
+// Read 100 bytes from the file.
+var readResult = byteChannel.read(100);
```

Comment:
Shall we rewrite the code to look like the following,
 
```
   // Here is how the bytes are read from the channel.
    var result = channel.read(100);
    match result {
        (blob, int) content => {
            return content;
        }
        error readError => {
            throw readError;
        }
    }
```

Reply:
IMO, its better to specially say that we are reading `100` bytes as `100` is passed as a parameter.
```ballerina
// Here is how 100 bytes are read from the channel.
var readResult = byteChannel.read(100);
    match readResult {
        (blob, int) content => {
            return content; // Return the read content.
        }
    error err => {
        return err; // An IO error occurred when reading the bytes.
    }
}
```
WDYT?

Reply:
+1